logstash-core/lib/logstash/logging/slow_logger/top_items.rb

```diff
@@ -0,0 +1,69 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "set"
+require 'thread'
+
+module LogStash; module Logging; module Util
+
+  class Operation
+
+    attr_reader :threshold, :time
+
+    def initialize(threshold, time)
+      @threshold = threshold
+      @time = time
+    end
+
+    def <=>(other)
+      if threshold != other.threshold
+        return threshold <=> other.threshold
+      end
+      return -1 if time > other.time
+      return  0 if time == other.time
+      return  1 if time < other.time
```

Comment:
As discussed previously at https://github.com/elastic/logstash/pull/5886#discussion_r78588439 the top item got removed with 52ed6bdd785448c8d829d054caf84eb7d34f64c1. I removed it basically because it's not used anymore in benefit of having a more general structure.